network-libp2p/src/swarm.rs

```diff
@@ -1204,6 +1222,29 @@ pub(crate) fn verify_record(record: &Record) -> Option<DhtRecord> {
     None
 }
 
+fn respond_on_behalf(
+    swarm: &mut NimiqSwarm,
+    request_id: InboundRequestId,
+    peer_id: PeerId,
+    type_id: RequestType,
+    channel: ResponseChannel<Option<Vec<u8>>>,
+    response: Result<(), InboundRequestError>,
+) {
+    if swarm
+        .behaviour_mut()
+        .request_response
+        .send_response(channel, Some(response.serialize_to_vec()))
+        .is_err()
+    {
+        error!(
+            %request_id,
+            %peer_id,
+            %type_id,
+            "Could not send response {:?}", response
+        );
+    }
+}
```

Comment:
it's repeated in two places close by and simplified a bit the code there because there are some many If and else's and nested scopes there. Can change it back if you prefer, either way is fine imo

Reply:
I removed the function and the response and respective error handling are done in the original places